dll/win32/shell32/CShellDispatch.cpp

```diff
@@ -119,58 +128,82 @@ HRESULT STDMETHODCALLTYPE CShellDispatch::BrowseForFolder(LONG Hwnd, BSTR Title,
 HRESULT STDMETHODCALLTYPE CShellDispatch::Windows(IDispatch **ppid)
 {
     TRACE("(%p, %p)\n", this, ppid);
+    return CoCreateInstance(CLSID_ShellWindows, NULL, CLSCTX_LOCAL_SERVER, IID_PPV_ARG(IDispatch, ppid));
+}
 
-    *ppid = NULL;
+static HRESULT SHELL_OpenFolder(LPCITEMIDLIST pidl, LPCWSTR verb = NULL)
+{
+    SHELLEXECUTEINFOW sei;
+    sei.cbSize = sizeof(sei);
+    sei.fMask = SEE_MASK_IDLIST | SEE_MASK_FLAG_DDEWAIT;
+    sei.hwnd = NULL;
+    sei.lpVerb = verb;
+    sei.lpFile = sei.lpParameters = sei.lpDirectory = NULL;
+    sei.nShow = SW_SHOW;
+    sei.lpIDList = const_cast<LPITEMIDLIST>(pidl);
+    if (ShellExecuteExW(&sei))
+        return S_OK;
+    DWORD error = GetLastError();
+    return HRESULT_FROM_WIN32(error);
+}
 
-    return E_NOTIMPL;
+static HRESULT OpenFolder(VARIANT vDir, LPCWSTR verb = NULL)
+{
+    CComHeapPtr<ITEMIDLIST> idlist;
+    HRESULT hr = VariantToIdlist(&vDir, &idlist);
+    if (hr == S_OK && SHELL_OpenFolder(idlist, verb) == S_OK)
```

Comment:
Shouldn't the result of `SHELL_OpenFolder` be checked with `SUCCEEDED()` instead of `== S_OK`?
Or is `S_FALSE` a failure here?

Reply:
Perhaps but you cannot open a folder window just a little bit, it makes no sense.

S_FALSE cannot happen with the current code.